src/datasets/data_files.py

```diff
@@ -46,36 +46,57 @@ class EmptyDatasetError(FileNotFoundError):
 }
 NON_WORDS_CHARS = "-._ 0-9"
 if config.FSSPEC_VERSION < version.parse("2023.9.0"):
-    KEYWORDS_IN_PATH_NAME_BASE_PATTERNS = ["{keyword}[{sep}/]**", "**[{sep}/]{keyword}[{sep}/]**"]
+    KEYWORDS_IN_FILENAME_BASE_PATTERNS = ["**[{sep}/]{keyword}[{sep}]*", "{keyword}[{sep}]*"]
```

Comment:
I think this is a breaking change impacting all uses of the variable `KEYWORDS_IN_PATH_NAME_BASE_PATTERNS`. See: https://github.com/huggingface/dataset-viewer/actions/runs/8753796799/job/24024224560?pr=2740

Reply:
`dataset-viewer` seems to be the only repo on GH using `KEYWORDS_IN_PATH_NAME_BASE_PATTERNS`...

But I think it's okay to add it back if it's hard to fix `datasets-viewer`'s CI otherwise.

Reply:
no need to add it back imo, we can just use the new variable in dataset-viewer

Reply:
I am fixing this in dataset-viewer: https://github.com/huggingface/dataset-viewer/pull/2740#issuecomment-2071741972